### ISSUE TITLE
Remove "delegateMinHeightActive" property from block headers

### ISSUE THREAD

**Issue by yatki** [LiskHQ/lisk-sdk#4413]
### Description

---

Currently, `delegateMinHeightActive ` property is part of block headers in BFT module:

https://github.com/LiskHQ/lisk-sdk/blob/e921d63ab800fb194402c63db716a6aa4199ca0b/framework/src/modules/chain/bft/bft.js#L31-L38

However, this information is not stored in the block. `delegateMinHeightActive ` can be calculated by using [`delegateMinHeightActive`](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0014.md#computing-prevotes-and-precommits) parameter.

`delegateMinHeightActive` must be removed from block headers and must be provided to related methods as an argument. Thus, we need to refactor the BFT module and update the BFT tests accordingly.

Therefore, BFT module should expect an interface as an input to the constructor
```
interface Forger {
  getActiveMinHeightActive: (stateStore: StateStore, publicKey: string, height: number, threshold?: number) => Promise<number>;
}
```
and this function should be implemented in the DPoS library

### Motivation

`activeSinceRound` is information not part of the block, thus shouldn't be passed in the block headers.

### Acceptance Criteria

- Refactor BFT module to have private functions to fetch `delegateMinHeightActive` from accounts.
- Refactor [FinalityManager.updatePrevotesPrecommits](https://github.com/LiskHQ/lisk-sdk/blob/e921d63ab800fb194402c63db716a6aa4199ca0b/framework/src/modules/chain/bft/finality_manager.js#L145) method to accept `delegateMinHeightActive` as an argument.

### Additional Information
* This parameter was renamed to `delegateMinHeightActive` with #4473

**Comment by yatki** [comment]
After discussing with @shuse2 we decided to pause this issue and tackle later it later. Because, if we can implement a better way to `undo` a block, we may not need to store the block headers in the first place.

So, for now `delegateMinHeightActive` information will remain to be injected into blockHeader object and used from there. (This also saves time since current implementation is already working like that.)

However, this issue is still valid, because designwise `delegateMinHeightActive` is not part of block headers.

**Comment by yatki** [comment]
Some of the work I've done for this issue can be found in `4413-remove-activeSinceRound branch`: https://github.com/LiskHQ/lisk-sdk/tree/4413-remove-activeSinceRound

**Comment by shuse2** [comment]
This should be done as part of using blocks cache

**Comment by shuse2** [comment]
This should be resolved together with #4737
